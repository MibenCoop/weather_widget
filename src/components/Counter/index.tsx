import React from 'react';

export interface Props {
    counter: number;
}
const Counter = ({ counter }: Props) => <p>{counter}</p>;
export default Counter;
