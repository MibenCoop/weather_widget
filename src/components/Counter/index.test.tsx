import React from 'react';
import { shallow, mount } from 'enzyme';

import Counter from './index';

describe('Counter component', () => {
    const nullCounterValue = 0;
    const counterValue = 10;
    it('should match snapshot with initial counter value', () => {
        const counter = shallow(<Counter counter={nullCounterValue} />);
        expect(counter).toMatchSnapshot();
    });
    it('should match snapshot with not null counter value', () => {
        const counter = shallow(<Counter counter={counterValue} />);
        expect(counter).toMatchSnapshot();
    });
    it('should be rendered correctly with given props', () => {
        const counter = mount(<Counter counter={counterValue} />);
        console.log('Counter length', counter.prop('counter'));
        expect(counter.prop('counter')).toEqual(counterValue);
    });
});
