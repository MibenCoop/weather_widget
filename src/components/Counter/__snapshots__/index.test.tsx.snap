// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should match snapshot with initial counter value 1`] = `
<p>
  0
</p>
`;

exports[`Counter component should match snapshot with not null counter value 1`] = `
<p>
  10
</p>
`;
